### PR TITLE
Add port checking when deleting vswitch

### PR DESCRIPTION
**Pull request by XiaowenJiang**
Signed-off-by: Julie Jiang <julie.jiang@emc.com>

### CODE REVIEW

**Comment by XiaowenJiang** [comment]
@weia5-emc @minweipan Please help us to review the code.

File: ivn/core.py [hunk 1]
```diff
@@ -423,18 +423,31 @@ def del_vswitch(self):
...
+        if ret == 0:
+            return ifname in out
+        else:
+            self.logger_topo.error(outerr)
+            return False
+
     def add_port(self, ifname):
         if not self.check_vswitch_exists():
             raise Exception("vswitch {} doesn't exist, please add it first.".format(self.name))
+        if self.check_port_exists(ifname):
```
Comment (weia5-emc) [inline ivn/core.py:437]: 👍

**Comment by XiaowenJiang** [comment]
@minweipan Please help us to review the changes. :trollface:

File: ivn/core.py [hunk 2]
```diff
@@ -448,6 +461,9 @@ def del_port(self, ifname):
     def del_all_ports(self):
         _, port_str, _ = start_process(["ovs-vsctl", "list-ports", self.name])
         port_list = port_str.split()
+        _, vir_port, _ = start_process(["ls", "/sys/devices/virtual/net"])
+        vir_port_list = vir_port.split()
+        port_list = set(vir_port_list).intersection(port_list)
```
Comment (weia5-emc) [inline ivn/core.py:466]: 👍

**Review by weia5-emc — APPROVED** [review]

**Pull request merged**